From e66eeb2496b743474c4c79aace7c601b7542e5db Mon Sep 17 00:00:00 2001
From: Hu Wang <huw@codeaurora.org>
Date: Thu, 8 Nov 2018 19:55:43 +0800
Subject: [PATCH 3/3] wifi: Introduce qti.hardware.wifi@1.0 HAL interface

This add vendor HAL interface to set wifi latency level.

CRs-Fixed: 2371274
Change-Id: I28dc1b02d1fe2e30a78a5cde0de988ab97c240f2
---
 wifi/1.0/IWifiVendorStaIface.hal | 50 ++++++++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)
 create mode 100644 wifi/1.0/IWifiVendorStaIface.hal

diff --git a/wifi/1.0/IWifiVendorStaIface.hal b/wifi/1.0/IWifiVendorStaIface.hal
new file mode 100644
index 0000000..ff25528
--- /dev/null
+++ b/wifi/1.0/IWifiVendorStaIface.hal
@@ -0,0 +1,50 @@
+/* Copyright (c) 2019, The Linux Foundation. All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are
+* met:
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above
+*       copyright notice, this list of conditions and the following
+*       disclaimer in the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of The Linux Foundation nor the names of its
+*       contributors may be used to endorse or promote products derived
+*       from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT
+* ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS
+* BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+* BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+* WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+* OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+* IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*/
+
+package vendor.qti.hardware.wifi@1.0;
+
+import android.hardware.wifi@1.0::types;
+import android.hardware.wifi@1.0::IWifiIface;
+import android.hardware.wifi@1.2::IWifiStaIface;
+
+/**
+ * Vendor StaIface Interface
+ */
+interface IWifiVendorStaIface extends IWifiStaIface{
+  /**
+   * Set latancy level to driver.
+   *
+   * @return status WifiStatus of the operation.
+   *         Possible status codes:
+   *         |WifiStatusCode.SUCCESS|,
+   *         |WifiStatusCode.ERROR_WIFI_IFACE_INVALID|,
+   *         |WifiStatusCode.UNKNOWN|
+   */
+  setLatencyLevel(uint32_t level) generates (WifiStatus status);
+};
-- 
2.17.1

